nshlib/nsh_parse.c

```diff
@@ -275,8 +276,11 @@ static const char g_line_separator[]  = "\"'#;\n";
 #ifdef CONFIG_NSH_ARGCAT
 static const char g_arg_separator[]   = "`$";
 #endif
-static const char g_redirect1[]       = ">";
-static const char g_redirect2[]       = ">>";
+static const char g_redirect_out1[]   = ">";
+static const char g_redirect_out2[]  = ">>";
+static const char g_redirect_in1[]   = "<";
+static const char g_redirect_in2[]  = "<<";
+static const char g_pipe[]  = "|";
```

Comment:
nice work, @casaroli do you plan to support pipe operator?

Reply:
Yes. I need the pipe operator. I think a simple limited implementation can be done, but it is tricky and require more refactor. Same goes for heredoc (`<<` redirection).

I kept it away from the current PR to avoid making it too big.

We can do it in the current PR or in a separate PR. WDYT?

Can we do in future PR but keep the checks in the parser?

Reply:
Yes, let's move g_pipe to new PR too.

Reply:
I removed `|` and `<<`. Those can be added in a future PR.